app/models/subscription.rb

```diff
@@ -69,4 +71,41 @@ def create_alert_run
   def expired?
     expires_on < Time.zone.today
   end
+
+  private
+
+  def needs_reference?
+    new_record? && reference.blank?
```

Comment:
This is probably moot, as `after_initialize` callback only gets called on a new record

Reply:
The Stack Overflow I read that told me how to do this suggested it. I will do more reading.

Reply:
According to https://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html, `after_initialize` is also called after a find. We don't want to add references to existing records, so I think this is required.